velox/vector/DictionaryVector.h

```diff
@@ -152,7 +152,20 @@ class DictionaryVector : public SimpleVector<T> {
   }
 
   BaseVector* loadedVector() override {
-    dictionaryValues_ = BaseVector::loadedVectorShared(dictionaryValues_);
+    if (initialized_) {
+      return this;
+    }
+
+    SelectivityVector rows(dictionaryValues_->size(), false);
+    for (vector_size_t i = 0; i < this->size(); i++) {
+      if (!BaseVector::isNullAt(i)) {
+        auto ind = getDictionaryIndex(i);
+        rows.setValid(ind, true);
+      }
+    }
+    rows.updateBounds();
+
+    LazyVector::ensureLoadedRows(dictionaryValues_, rows);
     setInternalState();
```

Comment:
My understanding is that initialize_ would be only set if isLazyNotLoaded() returns false which currently checks for allLoaded_ in lazyVector.

Reply:
Is that what we want, or not? I thought we want either both allLoaded_ and initialized_ to be set, or none, please correct me if I'm wrong here.

Reply:
@kgpai Krishna, I don't quite understand your comment. Would you clarify?